hydroflow_lang/src/graph/ops/mod.rs

```diff
@@ -58,6 +59,29 @@ pub enum PortListSpec {
     Fixed(Punctuated<PortIndex, Token![,]>),
 }
 
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone, Copy)]
+pub enum FlowPropertyVal {
+    Yes,
+    No,
+    Preserve,
+    CodeBlock,
+}
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone, Copy)]
+pub struct FlowProperties {
+    pub deterministic: FlowPropertyVal,
+    pub monotonic: FlowPropertyVal,
+    pub tainted: bool,
```

Comment:
`tainted` is probably too general a name. The intent is that it's tainted with "inconsistency" because something upstream is both non-deterministic and non-monotone. so could be `inconsistent` or `consistency-taint` or something.